wezterm-ssh/src/sessioninner.rs

```diff
@@ -279,20 +283,23 @@ impl SessionInner {
             #[cfg(unix)]
             unsafe {
                 use std::os::unix::io::{FromRawFd, IntoRawFd};
-                TcpStream::from_raw_fd(a.into_raw_fd())
+                Socket::from_raw_fd(a.into_raw_fd())
             }
             #[cfg(windows)]
             unsafe {
                 use std::os::windows::io::{FromRawSocket, IntoRawSocket};
-                TcpStream::from_raw_socket(a.into_raw_socket())
+                Socket::from_raw_socket(a.into_raw_socket())
             }
         } else {
-            let socket = TcpStream::connect((hostname.as_str(), port))
-                .with_context(|| format!("connecting to {}", remote_address))?;
-            socket
-                .set_nodelay(true)
-                .context("setting TCP NODELAY on ssh connection")?;
-            socket
+            let addr = (hostname.as_str(), port).to_socket_addrs()?.next().context("address")?;
```

Comment:
I think the context can be improved here to show what it was trying to do; in this case it is DNS resolving the hostname:

```suggestion
            let addr = (hostname.as_str(), port).to_socket_addrs()?.next().with_context(|| format!("resolving address for {}", hostname))?;
```